### ISSUE TITLE
conflict between DefaultCacheSerializer and ImageProcessor

### ISSUE THREAD

**Issue by gongruike** [onevcat/Kingfisher#1912]
### Check List

Thanks for considering to open an issue. Before you submit your issue, please confirm these boxes are checked.

- [x] I have read the [wiki page](https://github.com/onevcat/Kingfisher/wiki) and [cheat sheet](https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet), but there is no information I need.
- [x] I have searched in [existing issues](https://github.com/onevcat/Kingfisher/issues?utf8=✓&q=is%3Aissue), but did not find a same one.
- [x] I want to report a problem instead of asking a question. It'd better to use [kingfisher tag in Stack Overflow](http://stackoverflow.com/questions/tagged/kingfisher) to ask a question.

#### What

1. set property `preferCacheOriginalData` of DefaultCacheSerializer to true 
2. add a RoundCornerImageProcessor

when the first time retrieve image (from network), the result image is correct (with round corner), but if restart the app, the result image is incorrect (no round corner), because `preferCacheOriginalData ` is true, disk cache store the original data.

#### Reproduce
```
        let string = "remote image url"
        let source = Source.network(ImageResource(downloadURL: URL(string: string)!))

        let processor = RoundCornerImageProcessor(cornerRadius: 20)

        var serializer = DefaultCacheSerializer()
        serializer.preferCacheOriginalData = true

        let options = KingfisherManager.shared.defaultOptions + [.processor(processor)] + [.cacheSerializer(serializer)]

        button?.kf.setImage(with: source, for: .normal, placeholder: nil, options: options) { receivedSize, totalSize in
            print("\(receivedSize)_\(totalSize)")
        } completionHandler: { result in
            switch result {
            case .success(let image):
                print(image)
            case .failure(let error):
                print(error)
            }
        }
```

#### Other Comment

kingfisher 7.1.2
Xcode 13.2.1

**Comment by onevcat** [comment]
Are you trying to load a JPEG image which is not containing an alpha channel? The default cache serializer would prefer to keep the image format so it might lose the alpha information if the original one is JPEG. You can add a `.png` formatter to solve it. See the [wiki here](https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet#serializer-to-force-a-format)

**Comment by gongruike** [comment]
> Are you trying to load a JPEG image which is not containing an alpha channel? The default cache serializer would prefer to keep the image format so it might lose the alpha information if the original one is JPEG. You can add a `.png` formatter to solve it. See the [wiki here](https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet#serializer-to-force-a-format).

It's not about the alpha channel. Replace `RoundCornerImageProcessor` with `BlurImageProcessor` or `BorderImageProcessor` will get the same problem. Because set `preferCacheOriginalData` to true, disk cache will store the original image data instead of the processed image data.

**Comment by onevcat** [comment]
Oh, I see, you are right on this. I will see how to fix this.

May I know what is your purpose of setting `preferCacheOriginalData`? If you just want to cache the original image data, maybe try adding `.cacheOriginalImage` to the option when setting image. It would help any other following processing to skip downloading the same image data again.

**Comment by gongruike** [comment]
> Oh, I see, you are right on this. I will see how to fix this.
> 
> May I know what is your purpose of setting `preferCacheOriginalData`? If you just want to cache the original image data, maybe try adding `.cacheOriginalImage` to the option when setting image. It would help any other following processing to skip downloading the same image data again.

I haven't use `preferCacheOriginalData`.  I was reading the source code and got a little confuse on that part.

**Comment by onevcat** [comment]
Solved in #1999
